### PR TITLE
test: add a case for ``C, enh: consider name after "struct", "union", or "enum"''

### PR DESCRIPTION
**Pull request by masatake**

### CODE REVIEW

File: tests/lang-c/test.el [hunk 1]
```diff
@@ -1,18 +1,57 @@
+(defun adjust-pattern (pattern)
```
Comment (AmaiKinono) [inline tests/lang-c/test.el:1]: I think you could reuse `citre-core--parse-search-pattern` here.

File: tests/lang-c/test.el [hunk 2]
```diff
@@ -1,18 +1,57 @@
...
-      (point-min) (point-max)))))
+  (let (last-abspath)
+    (with-temp-buffer
+      (mapc (lambda (def)
+	      (let ((abspath (gethash 'ext-abspath def)))
+		(unless (equal abspath last-abspath)
+		  (insert abspath)
+		  (insert ?\n)
+		  (setq last-abspath abspath)))
+	      (insert (gethash 'line def))
```
Comment (AmaiKinono) [inline tests/lang-c/test.el:31]: Can we use `citre-make-tag-str` here? I think we could assemble a string in xref style with it, using these components: `'(location :no-path t :suffix ":")`, `'(annotation :prefix "(" :suffix ")")` and `'(content)`.

Comment (masatake) [reply tests/lang-c/test.el]: Thank you. With citre-make-tag-str, we can remove adjust-pattern!

**Comment by AmaiKinono** [comment]
Thanks!

**Pull request merged**